src/plugins/gs-plugin-external-apps.c

```diff
@@ -540,9 +563,7 @@ extract_runtime_info_from_json_data (const char *data,
 
 	/* optional elements */
 	type_str = json_object_get_string_member (runtime, JSON_RUNTIME_TYPE_KEY);
-	if (type_str) {
-		*type = g_strdup (type_str);
-	}
+	*type = g_strdup (type_str);
```

Comment:
This fix also doesn't belong to this commit.
If you need help with git sorcery let me know on Slack. I'll be happy to help.